src/System.Runtime.Extensions/tests/System/Environment.Exit.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using Xunit;
+
+public class Environment_Exit
+{
+    [Fact]
+    public static void CheckExitCode()
+    {
+        RunRemote(0);
+        RunRemote(1);
+        RunRemote(42);
+        RunRemote(-1);
+        RunRemote(-45);
+    }
+
+    private static void RunRemote(int arg)
+    {
+        using (Process p = new Process())
+        {
+            try
+            {
+                ProcessStartInfo psi = new ProcessStartInfo();
+                psi.FileName = "corerun";
+                psi.Arguments = "EnvironmentExitConsoleApp.exe " + Convert.ToString(arg);
+                psi.UseShellExecute = false;
```

Comment:
> since the API uses reflection to invoke the lambda the exit code was not getting propagated correctly

If that's still happening even after recent coreclr fixes, than we should create a repro and open a bug for it.

> I don't want to block on it

Ok, but then please open an issue for it so we don't forget to follow up and remove this.